lib/rules/selector-max-specificity/index.mjs

```diff
@@ -246,19 +246,21 @@ const rule = (primary, secondaryOptions) => {
 						continue;
 					}
 
-					parseSelector(resolvedSelector, result, ruleNode, (selectorTree) => {
-						// Check if the selector specificity exceeds the allowed maximum
-						if (compare(maxChildSpecificity(selectorTree), maxSpecificity) > 0) {
-							report({
-								ruleName,
-								result,
-								node: ruleNode,
-								message: messages.expected,
-								messageArgs: [resolvedSelector, primary],
-								word: selector,
-							});
-						}
-					});
+					const selectorTree = parseSelector(resolvedSelector, result, ruleNode);
```

Comment:
I don't like callbacks here because they hide a `if` statement. Checking a return value is more explicit, and these refactoring at this level is acceptable, I think.